sootty/storage/wire.py

```diff
@@ -138,9 +143,10 @@ def __rshift__(self, other):
         return wire
 
     def __add__(self, other):
-        wire = Wire(name="(" + self.name + " + " + other.name + ")")
-        wire._data = self._data.__add__(other._data)
-        return wire
+        if isinstance(other, Wire):
```

Comment:
We shouldn't be throwing type errors this deep in the code; The users of the library aren't meant to interact directly with wire objects, and adding logic to handle mismatched types just adds more complexity to the code. I think it's fine to leave it out.